recipes-openxt/argo-module/argo-module/linux-argo-xmit-xsm-context.patch

```diff
@@ -0,0 +1,206 @@
+From 9b019db98c2c52d9c34b1734e82b878417a3c89c Mon Sep 17 00:00:00 2001
+From: Christopher Clark <christopher.w.clark@gmail.com>
+Date: Wed, 14 Aug 2019 14:56:35 -0700
+Subject: [PATCH] OXT-1503: Update the argo-linux driver to new Argo ABI with
+ XSM sid in message header
+
+Handles the changes to the Argo message header format and sendv operation
+arguments for the ABI change to indicate XSM/Flask SID in Argo messages:
+the message_type field has been reduced in size to 8 bits from 32 bits,
+and the message source domain and port fields are now standalone rather
+than within a xen_argo_addr struct.
+
+Since the message_type field, which is used to indicate the protocol that
+is in use between driver endpoints -- ie. either datagram or stream -- has
+decreased in size, the protocol indicator values for dgram and stream are
+changed to new values within the 8-bit range.
+
+Signed-off-by: Christopher Clark <christopher.clark6@baesystems.com>
+
+ argo-module.c      | 43 ++++++++++++++++++++-----------------------
+ include/xen/argo.h | 11 +++++++----
+ 2 files changed, 27 insertions(+), 27 deletions(-)
+
+--- a/argo-module.c
++++ b/argo-module.c
+@@ -236,8 +236,8 @@ static struct list_head ring_list;
+ #define ARGO_SHF_PING        (1 << 8)
+ #define ARGO_SHF_PONG        (1 << 9)
+ 
+-#define ARGO_PROTO_DGRAM     0x6447724d
+-#define ARGO_PROTO_STREAM    0x3574526d
++#define ARGO_PROTO_DGRAM     0x01
++#define ARGO_PROTO_STREAM    0x02
```

Comment:
This doesn't have to be addressed in this patch, but you now have ARGO_PROTO_DGRAM=1 & ARGO_PROTO_STREAM=2 here and ARGO_PTYPE_DGRAM=1 & ARGO_PTYPE_STREAM=2, in argo-linux/include/linux/argo_dev.h.  Can ARGO_PROTO_DGRAM/STREAM just be converted to ARGO_PTYPE_*.  It's not clear why argo_dev.h exposes ptype.

Also, looking in /usr/include/bits/socket_type.h the defines are flipped compared to argo:
SOCK_STREAM = 1
SOCK_DGRAM = 2

Reply:
To pile on, would there be a reason we just drop "ARGO_PROTO_*" and use the "standard" [socket types](https://elixir.bootlin.com/linux/latest/source/include/linux/net.h#L59)?

Reply:
Looking at it... so prior to this change there was a clear distinction between the magic-valued protocol indicator that went between domains (ARGO_PROTO_DGRAM/STREAM) and the protocol indicator that goes between userspace and the kernel (ARGO_PTYPE_DGRAM/STREAM),
eg. supplied in the argument to bind or returned by the ioctl.

After this PR which effectively removes the magic values, yep, it looks like using standard `sock_type` values for both would be fine. There's some conversion logic in `libargo` which converts between ARGO_PTYPE and SOCK values which could go too.

It gets slightly more painful to do the above in a patch carried within `xenclient-oe` because we have several argo recipes as part of resolving our exotic circular dependencies in that layer, and it would touch multiple of them. That points towards running a branch on the `linux-xen-argo` repository for OpenXT to do this work.